### DISCUSSION TITLE
How to retrieve the Name of activities in Elsa Workflow?

### DISCUSSION THREAD

**Discussion by vinhlam1412** [elsa-workflows/elsa-core#4515] [Q&A]
This my table in database: 

 
![1](https://github.com/elsa-workflows/elsa-core/assets/91020638/877be909-676c-4f76-95f2-8b3dc889808a)

 

I have a Document, when I new a Document, it’s Status default is Draft. It’s have 2 Action is: Submit and Cancel 

 
![2](https://github.com/elsa-workflows/elsa-core/assets/91020638/72e8de0f-54ff-4c2f-8a89-234a7e3084d3)

 

 

When I choose Submit, Status will change to NextState(Posted) in table above , and Action is Cancel 

 
![3](https://github.com/elsa-workflows/elsa-core/assets/91020638/20657ef7-423b-4817-b4c3-78e8234a3a45)

 

I want use ElsaWorkflow dashboard to implement: 

If I want to load those actions from the workflow without loading them from a table, how would I do that (do I need to use Signal Received or State)? 

It would be like this: 

 
![4](https://github.com/elsa-workflows/elsa-core/assets/91020638/62a90487-3943-41d0-b5f9-06700194f1ca)

And I want to load the Action button based on the Status from the red areas, how do I do that? For example, if the Status is "Draft," the Action button will have a dropdown with "Submit" and "Cancel" options.

**Comment by sfmskywalker** [comment]
Got it. You will want to use the UserTask activity to achieve this. It works similar to the Fork activity, but with the added benefit that the UserTask activity stores the available actions in a database table that your application can query. Specifically, you would be loading the available actions for a given workflow instance from the Bookmarks table, by selecting all rows that match the workflow instance ID and the UserTask activity type name. The matching bookmark contains a payload created by the UserTask activity, which contains the available actions.

You can then render these actions in your UI as you have depicted above.
When the user selects a given action, you can communicate that back to the workflow engine, which will then resume the UserTask activity and continue with the outcome that matches the selected user action.

Take a look at the `IUserTaskService` service in the Elsa.Activities.UserTask project's source code for available methods.
Unfortunately, there's no REST API (yet) that you could invoke directly, so you'll have to implement your own API controllers that your application can invoke.

**Reply by vinhlam1412** [reply]
Thank you for your answer. I would like to ask one more thing. I'm working entirely on the Elsa dashboard, so how do I retrieve the workflow instance ID? Could you provide me with an example code?
This is my workflow with Endpoint
![UserTask](https://github.com/elsa-workflows/elsa-core/assets/91020638/bfd83cb4-5c8b-442c-9772-5df59135efce)

**Reply by sfmskywalker** [reply]
Sure, if you need access to the current workflow instance ID from a JavaScript expression, you can do this:

```javascript
workflowInstanceId
```

You can use this in activities such as HTTP Response if you wanted to send the workflow instance ID to the client.
You can find more J variables and functions [here](https://v2.elsaworkflows.io/docs/expressions/javascript#workflow-instance-id).

**Reply by vinhlam1412** [reply]
Can I retrieve the Name of each Case? I want to save this value in a database and then call the update API (changing the Status from Draft to Posted). How can I accomplish that? Thank you very much.
![switch](https://github.com/elsa-workflows/elsa-core/assets/91020638/c7bac2e3-b30a-48c5-9f7d-657e1ec4dbb6)
![switch2](https://github.com/elsa-workflows/elsa-core/assets/91020638/8ff4ba30-c408-4972-b8a6-05c58ed8d2dc)

**Reply by sfmskywalker** [reply]
I'm sure it's possible, but it's possible that we might have to add some code in order to make it accessible. If you could elaborate a little bit on where you need to access the case name, I will have better context in order to provide you with a solution.
